CONTRIBUTING.md

```diff
@@ -1,112 +1,124 @@
+meson compile srpm -C builddir
+mock --rebuild ./builddir/dist/srpm/*.src.rpm
 ```
 
 # Test environment Quickstart
 
-This environment recreates the `rhc-worker-playbook`, in which an `ansible playbook` is sent through the MQTT broker, `yggdrasil` processes it and send it to the worker to execute it.
+This environment recreates the `rhc-worker-playbook`, in which an `ansible
+playbook` is sent through the MQTT broker, `yggdrasil` processes it and send it
+to the worker to execute it.
 
 
-**TERMINAL 1**
+## TERMINAL 1
 
 Run an *HTTP server* that will serve the payload. Then run `mosquitto` service.
 
 ```console
-$ nohup python3 -m http.server 8000 > /dev/null 2>&1 &
-$ mosquitto
+$ nohup python3 -m http.server --directory ./testdata 8000 > /dev/null 2>&1 &
+$ systemctl start mosquitto
 ```
 
-**TERMINAL 2**
+## TERMINAL 2
 
-Run `yggdrasil` specifying `mosquitto` as the MQTT server.
+Configure yggdrasil to connect to the local `moquitto` broker and increase the
+log level.
 
 ```console
-$ sudo go run ./cmd/yggd --server tcp://localhost:1883 --log-level trace --socket-addr @yggd
-...
-subscribed to topic: yggdrasil/localhost-22278168-85a6-11ec-a65c-fa163e3b5a61/data/in
-...
+cat /etc/yggdrasil/config.toml
+
+protocol = "mqtt"
+server = ["tcp://localhost:1883"]
+log-level = "debug"
 ```
-Or, if it is using via `rhc`.
 
-Configure in rhc the `moquitto` connection and the log level.
+Start the service:
 
 ```console
-cat /etc/rhc/config.toml
-# rhc global configuration settings
-
-broker = ["tcp://localhost:1883"]
-cert-file = "/etc/pki/consumer/cert.pem"
-key-file = "/etc/pki/consumer/key.pem"
-log-level = "bug"
+sudo systemctl start yggdrasil.service
 ```
 
-**TERMINAL 3**
+## TERMINAL 3
 
-Associate the `@yggd` socket with the `rhc-worker-playbook` worker.
+Assuming you built and installed the `rhc-worker-package` as described in
+[#Compilation](#compilation), start
+`com.redhat.Yggdrasil1.Worker1.rhc_worker_playbook.service`:
 
 ```console
-sudo YGG_SOCKET_ADDR=unix:@yggd python3 /usr/libexec/rhc/rhc-worker-playbook.worker
+sudo systemctl start com.redhat.Yggdrasil1.Worker1.rhc_worker_playbook.service
 ```
-This step is not needed if using the worker through `rhc`.
 
-**TERMINAL 4**
+## TERMINAL 4
 
-Execute the publish message in the "control/in" topic. Using `yggctl` to generate the data message with the Ansible playbook `test-module.yml` served by the HTTP server and `rhc-worker-playbook` as the directive.
+Generate a data message and publish it. `yggctl` (installed as part of the
+`yggdrasil` package) can easily generate data messages in the JSON format
+expected by `yggd`.
 
 ```console
- go run ./cmd/yggctl generate data-message --directive "rhc-worker-playbook" \"http://localhost:8000/test-module.yml\" | pub -broker tcp://localhost:1883 -topic yggdrasil/$CLIENT_ID/data/in
- ```
+echo '"http://localhost:8000/insights_remove.yml"' | \
+  yggctl generate data-message --directive rhc_worker_playbook -
+```
```

Comment:
When I run this command, then I get following result:

```
echo '"http://localhost:8000/insights_remove.yml"' |   yggctl generate data-message --directive rhc_worker_playbook -
{"type":"data","message_id":"28ab2d71-1077-48fd-bd67-c496cd9a5015","response_to":"","version":1,"sent":"2024-08-12T23:06:28.615688017+02:00","directive":"rhc_worker_playbook","metadata":{},"content":"Imh0dHA6Ly9sb2NhbGhvc3Q6ODAwMC9pbnNpZ2h0c19yZW1vdmUueW1sIgo="}
```

Reply:
This is caused by a bug in older versions of yggctl. Make sure you're using
0.4.2 or later.